lib/widgets/like_button.dart

```diff
@@ -0,0 +1,121 @@
+import 'package:flutter/widgets.dart';
+import 'dart:math';
+
+import '../common/colors.dart';
+
+class LikeButton extends StatefulWidget {
+  final bool selected;
+  final double width;
+  final double height;
+  final void Function(bool) onChange;
+
+  LikeButton({
+    this.selected = false,
+    this.width = 20,
+    this.height = 20,
+    this.onChange,
+  });
+
+  @override
+  _LikeButtonState createState() => _LikeButtonState();
+}
+
+class _LikeButtonState extends State<LikeButton>
+    with SingleTickerProviderStateMixin {
+  Animation<double> _animation;
+  AnimationController controller;
+  double percent = 0.0;
+
+  bool _selected;
+
+  @override
+  void initState() {
+    super.initState();
+
+    _selected = widget.selected;
+    if (_selected) {
+      percent = 1.0;
+    }
+
+    controller =
+        AnimationController(duration: Duration(milliseconds: 250), vsync: this)
+          ..value = percent;
+  }
+
+  @override
+  void dispose() {
+    controller.dispose();
+    super.dispose();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    _animation = CurvedAnimation(parent: controller, curve: Curves.easeIn)
+      ..addListener(() {
+        setState(() {
+          percent = _animation.value;
+        });
+      });
+
+    return Container(
+        width: widget.width,
+        height: widget.height,
+        decoration:
+            BoxDecoration(border: Border.all(color: Colors.likeButtonColor)),
+        child: GestureDetector(
+            onTap: () {
+              if (!_selected) {
+                controller.forward();
+              } else {
+                controller.reverse();
+              }
+
+              if (widget.onChange != null) {
+                widget.onChange(!_selected);
+              }
+
+              setState(() {
+                _selected = !_selected;
+              });
+            },
```

Comment:
These lines could be extracted to a method for better readability of Widget code.